### PR TITLE
[Elfutils_jll] Bump to 0.182

### PR DESCRIPTION
**Pull request by jpsamaroo**
So that we pickup libobstack.so.1 instead of libobstack.so.0

### CODE REVIEW

**Comment by jpsamaroo** [comment]
@vchuravy

**Comment by giordano** [comment]
> So that we pickup libobstack.so.1 instead of libobstack.so.0

Does this mean products in different versions of obstack_jll have different ABIs?

**Comment by jpsamaroo** [comment]
I honestly have no idea, it just can't find libobstack.so since it's linked against the major version.

**Comment by giordano** [comment]
Uhm, yes: https://github.com/JuliaBinaryWrappers/obstack_jll.jl/commit/7f044a88d1e00a59d9d7438b1de512b35f57bc27#diff-f1ba1a514600ecf25768f83d6258f7138480b9e3d4ab07a61666bf403819000e.  This means `obstack_jll` 1.2.2 has a different ABI from 1.1.0, and you should really specify the compat for that dependency.  However, changing compat without bumping the version number is very bad, as it'll screw things up for people who have the package installed now, or if they try to instantiate a manifest.

**Comment by jpsamaroo** [comment]
So what needs to be done for this PR?

**Comment by giordano** [comment]
Change the version number somehow (is there a newer upstream version) and add a compat for `obstack_jll`, something like
```julia
Dependency("obstack_jll"; compat="~1.2.2"),
```
(I don't know what the compat specifier should be, but they broke ABI-compatibility within a minor version.....)

**Comment by giordano** [comment]
There is 0.185 now: https://sourceware.org/elfutils/ftp/0.185/.  While you're there, you can request Julia v1.6, build for experimental platforms and use
```
Dependency("Bzip2_jll"; compat="1.0.8")
```
as dependency

**Comment by jpsamaroo** [comment]
I don't know how to request Julia 1.6, but I did the rest of the things you recommended.

File: E/Elfutils/build_tarballs.jl [hunk 1]
```diff
@@ -57,12 +57,13 @@ dependencies = [
...
     Dependency("XZ_jll"),
     Dependency("argp_standalone_jll"),
     Dependency("fts_jll"),
-    Dependency("obstack_jll"),
+    Dependency("obstack_jll"; compat="~1.2.2"),
 ]
 
+
 # Build the tarballs, and possibly a `build.jl` as well.
 build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)
```
Comment (giordano) [inline E/Elfutils/build_tarballs.jl:69]:
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6")
```

File: E/Elfutils/build_tarballs.jl [hunk 2]
```diff
@@ -43,7 +43,7 @@ install_license COPYING*
 """
 
 # Only build for Linux
-platforms = [p for p in supported_platforms() if Sys.islinux(p)]
+platforms = [p for p in supported_platforms(; experimental=true) if Sys.islinux(p)]
```
Comment (giordano) [inline E/Elfutils/build_tarballs.jl:46]:
```suggestion
platforms = filter!(Sys.islinux, supported_platforms(; experimental=true))
```

**Comment by jpsamaroo** [comment]
Alpine has a working build for 0.182, so I switched us to that.

**Comment by jpsamaroo** [comment]
That failing build is hopefully fixed by https://github.com/JuliaPackaging/Yggdrasil/pull/3350

**Comment by jpsamaroo** [comment]
@giordano I'm not sure what I'm doing wrong here.

**Comment by giordano** [comment]
I'm not really sure what's going with argp_standalone, but we're also missing fts, so maybe let's skip experimental platforms for the time being :slightly_smiling_face:

**Comment by jpsamaroo** [comment]
@giordano the Elfutils wrappers don't seem to have updated? They're still on 0.179

**Comment by giordano** [comment]
Where?  The registry is up-to-date: https://github.com/JuliaRegistries/General/commit/7725b9b5479a3708c362ce0eef67f1630302bda1

**Comment by jpsamaroo** [comment]
Huh, weird, seems like my system is having trouble picking 0.182 over 0.179 automatically.

**Pull request merged**